src/scikit_build_core/program_search.py

```diff
@@ -77,14 +78,14 @@ def get_cmake_program(cmake_path: Path) -> Program:
     None if it cannot be determined.
     """
     try:
-        result = Run().capture(cmake_path, "--version")
+        result = Run().capture(cmake_path, "-E", "capabilities")
     except (subprocess.CalledProcessError, PermissionError):
         return Program(cmake_path, None)
 
     try:
-        version = Version(result.stdout.splitlines()[0].split()[-1].split("-")[0])
-    except (IndexError, InvalidVersion):
-        logger.warning(f"Could not determine CMake version, got {result.stdout!r}")
+        version = Version(json.loads(result.stdout)["version"]["string"])
+    except (json.decoder.JSONDecodeError, KeyError, InvalidVersion):
+        logger.warning("Could not determine CMake version, got {!r}", result.stdout)
```

Comment:
```suggestion
        logger.warning(f"Could not determine CMake version, got {result.stdout!r}")
```
Preferably, keep the f-strings

Reply:
`logger.warning()` takes a format string and applies arguments to it. In practice this was never a problem until I changed one of the test cases to output `{}`, which gets mistaken for a format placeholder.

In any case, I really think that when a function takes a format string and format arguments, we should give it those rather than an f-string, otherwise it's running the format algorithm twice and we have to escape instances of `{}`.

Reply:
Hmm, indeed I see that everywhere else argument style is used and only here the is an f-string. Not sure what format @henryiii prefers for this. I think having it f-string in place is more readable, and I don't think there are measurable performance difference there. Anyway let's see what @henryiii prefers here

Reply:
The logger doesn't build the string (or stringify the args) if it's not logging, so avoiding the f-string is better for logs.